server/grpc_service.go

```diff
@@ -513,13 +507,101 @@ func (s *GrpcServer) forwardTSO(stream pdpb.PD_TsoServer) error {
 	}
 }
 
+func (s *GrpcServer) forwardTSORequestWithDeadLine(
+	ctx context.Context, request *pdpb.TsoRequest, forwardStream tsopb.TSO_TsoClient,
+) (*tsopb.TsoResponse, error) {
+	defer logutil.LogPanic()
+	// Create a context with deadline for forwarding TSO request to TSO service.
+	ctxTimeout, cancel := context.WithTimeout(ctx, tsoutil.DefaultTSOProxyTimeout)
+	defer cancel()
+
+	tsoProxyBatchSize.Observe(float64(request.GetCount()))
```

Comment:
Do we need this metrics?

Reply:
Yes? to be consistent with the tso dispaching, otherwise this metrics will shows no value which might cause confusing.

Reply:
This is exsitng metrics, so it's meaningful even just show constant value 1 instead of 0 or no value on the dashboard.

Reply:
We don't have the batch logic here, so I think it's unnecessary.